File: examples/text-to-speech/js/websocket/twilio/test_call.js
require('dotenv').config();
const twilio = require('twilio');

const accountSid = process.env.TWILIO_ACCOUNT_SID;
const apiKeySid = process.env.TWILIO_API_KEY_SID;
const apiKeySecret = process.env.TWILIO_API_KEY_SECRET;

const client = twilio(apiKeySid, apiKeySecret, { accountSid });

const TO_NUMBER = process.env.TO_NUMBER;
const FROM_NUMBER = process.env.FROM_NUMBER;
const URL = process.env.NGROK_URL ? `${process.env.NGROK_URL}/voice` : 'http://localhost:3000/voice';

console.log(`Initiating call from ${FROM_NUMBER} to ${TO_NUMBER}`);
console.log(`Webhook URL: ${URL}`);

client.calls
  .create({
    url: URL,
    to: TO_NUMBER,
    from: FROM_NUMBER,
  })
  .then(call => console.log(`Call started: ${call.sid}`))
  .catch(err => console.error(`Error: ${err.message}`));

File: examples/text-to-speech/python/websocket/twilio/test_call.py
from twilio.rest import Client
from src.config import (
    TWILIO_ACCOUNT_SID,
    TWILIO_API_KEY_SID,
    TWILIO_API_KEY_SECRET,
    NGROK_URL,
    PORT,
    FROM_NUMBER,
    TO_NUMBER
)


def main():
    """Make a test call using Twilio."""
    # Create Twilio client with API key authentication
    client = Client(TWILIO_API_KEY_SID, TWILIO_API_KEY_SECRET, TWILIO_ACCOUNT_SID)

    # Determine the webhook URL
    base_url = NGROK_URL if NGROK_URL else f"http://localhost:{PORT}"
    webhook_url = f"{base_url}/voice"

    print(f"Making test call from {FROM_NUMBER} to {TO_NUMBER}")
    print(f"Webhook URL: {webhook_url}")

    try:
        call = client.calls.create(
            to=TO_NUMBER,
            from_=FROM_NUMBER,
            url=webhook_url
        )
        print(f"Call initiated successfully! Call SID: {call.sid}")
    except Exception as e:
        print(f"Failed to make call: {e}")


if __name__ == "__main__":
    main()
